### PR TITLE
Use list api to determine whether an app service exists instead of Exception

### PR DESCRIPTION
**Pull request by Flanker32**
Use list api to determine whether an app service exists instead of Exception

### CODE REVIEW

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/AbstractFunctionMojo.java [hunk 1]
```diff
@@ -151,15 +150,13 @@ public String getFinalName() {
...
-            return getAzureClient().appServices().functionApps().getByResourceGroup(getResourceGroup(), getAppName());
-        } catch (AzureAuthFailureException authEx) {
-            throw authEx;
-        } catch (Exception ex) {
-            Log.debug(ex);
-            // Swallow exception for non-existing Azure Functions
-        }
-        return null;
+        final PagedList<FunctionApp> functionList = getAzureClient().appServices().functionApps().list();
+        functionList.loadAll();
```
Comment (andxu) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/AbstractFunctionMojo.java]: check whether loadAll will load all heavy properties for each app

Comment (Flanker32) [reply azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/AbstractFunctionMojo.java]: Change to use iterator

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/AbstractFunctionMojo.java [hunk 2]
```diff
@@ -151,13 +151,15 @@ public String getFinalName() {
...
-        } catch (AzureAuthFailureException authEx) {
-            throw authEx;
-        } catch (Exception ex) {
-            Log.debug(ex);
-            // Swallow exception for non-existing Azure Functions
+        final PagedList<FunctionApp> functionList = getAzureClient().appServices().functionApps().list();
+        final ListIterator<FunctionApp> functionAppListIterator = functionList.listIterator();
+        FunctionApp functionApp = null;
+        while (functionAppListIterator.hasNext()) {
+            functionApp = functionAppListIterator.next();
```
Comment (andxu) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/AbstractFunctionMojo.java]: duplicate code with webapp

Comment (Flanker32) [reply azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/AbstractFunctionMojo.java]: Extracted it to AppServiceUtils.

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/utils/AppServiceUtils.java [hunk 3]
```diff
@@ -126,4 +128,10 @@ public static DockerImageType getDockerImageType(final String imageName, final S
             return isPrivate ? DockerImageType.PRIVATE_DOCKER_HUB : DockerImageType.PUBLIC_DOCKER_HUB;
         }
     }
+
+    public static <T extends WebAppBase> T findAppServiceInPagedList(PagedList<T> list, String resourceGroup, String name) {
+        final Iterator<T> iterator = list.listIterator();
```
Comment (andxu) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/utils/AppServiceUtils.java]: need to check list for null and resourceGroup/name for non-empty

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java [hunk 4]
```diff
@@ -81,7 +81,7 @@
             "\n\nYou can change your telemetry configuration through 'allowTelemetry' property.\n" +
             "For more information, please go to https://aka.ms/azure-maven-config.\n";
     public static final String INVALID_AUTH_TYPE = "%s is not a valid auth type for Azure maven plugins, " +
-            "supported values are %s. Will use AUTO by default";
+            "supported values are %s. Will use auto by default";
```
Comment (andxu) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java]: 'auto'

Comment (Flanker32) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java]: Fixed

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/utils/AppServiceUtils.java (hunk 3 above)
Comment (Flanker32) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/utils/AppServiceUtils.java]: Added

**Pull request merged**